core/descriptors/src/org/jetbrains/kotlin/types/AbstractClassTypeConstructor.java

```diff
@@ -23,14 +23,14 @@
 import org.jetbrains.kotlin.resolve.DescriptorUtils;
 
 public abstract class AbstractClassTypeConstructor implements TypeConstructor {
-    private boolean hashCodeComputed;
+    private volatile boolean hashCodeComputed;
     private int hashCode;
 
     @Override
     public final int hashCode() {
```

Comment:
@yole, Maybe there would be best to use String.hashCode() approach? It doesn't require additional volatile field.